utils/serverStatistic.js

```diff
@@ -0,0 +1,130 @@
+    const regular = getStatsObject(statsNames, tankStats.all);
+    result[tankStats.tank_id] = { regular, battleLifeTime };
+  });
+
+  achievements.map((tankAchivements) => {
+    if (!result[tankAchivements.tank_id]) return;
+    const mastery = getStatsObject(achievementsName, tankAchivements.achievements);
+    result[tankAchivements.tank_id].mastery = mastery;
+  });
+
+  return Object.entries(result).map(([wotId, data]) => ({ wotId, ...data }));
+};
+
+const fetchPlayerStats = async (playerIds) => {
+  const playerStatsURL = `${BASE_URL}/account/info/?application_id=${application_id}&account_id=${playerIds.join(',')}&extra=statistics.rating`;
+  const playerAchievmentsURL = `${BASE_URL}/account/achievements/?application_id=${application_id}&account_id=${playerIds.join(',')}`;
+
+  try {
+    const playerStats = await getInfo(playerStatsURL);
+    const playerAchievments = await getInfo(playerAchievmentsURL);
+
+    for (const accountId in playerStats) {
+      const stats = playerStats[accountId]?.statistics;
+      const achievements = playerAchievments[accountId]?.achievements;
+      const vehicleStats = await getVehicleStats(accountId);
+
+      if (!stats || !achievements) continue;
+      const regularStats = stats.all;
+      const ratingStats = stats.rating;
+
+      if (JSON.stringify(allPlayerStats.account) === '{}') {
+        allPlayerStats.account.regular = getStatsObject(statsNames, regularStats);
+        allPlayerStats.account.rating = getStatsObject(statsNames, ratingStats);
+        allPlayerStats.account.mastery = getStatsObject(achievementsName, achievements);
+      } else {
+        statsNames.map((statName) => {
+          allPlayerStats.account.regular[statName] = allPlayerStats.account.regular[statName] + regularStats[statName];
+          allPlayerStats.account.rating[statName] = allPlayerStats.account.rating[statName] + ratingStats[statName];
+        });
+        achievementsName.map((achievementName) => {
+          allPlayerStats.account.mastery[achievementName] = allPlayerStats.account.mastery[achievementName] + (achievements[achievementName] || 0);
+        });
+      }
+
+      if (!allPlayerStats.tanks.length && vehicleStats) allPlayerStats.tanks.push(...vehicleStats);
+      else if (vehicleStats) {
+        vehicleStats.map((vehicle) => {
+          const { wotId, regular, battleLifeTime, mastery } = vehicle;
+          if (allPlayerStats.tanks.find((tank) => tank.wotId === wotId)) {
+            const existingTank = allPlayerStats.tanks.find((tank) => tank.wotId === wotId);
+            statsNames.map((statName) => {
+              existingTank.regular[statName] += regular[statName];
+            });
+            achievementsName.map((achievementName) => {
+              existingTank.mastery[achievementName] += mastery[achievementName] || 0;
+            });
+          } else {
+            allPlayerStats.tanks.push({ wotId, regular, battleLifeTime, mastery });
+          }
+        });
+      } 
```

Comment:
The if condition `!allPlayerStats.tanks.length && vehicleStats` is no longer necessary. Also you don’t adding `battleLifeTime` to your existing tank statistic, so the number didn't change even if the stat to add was. You can change `for…of` instead of `map`, since you don’t return any array.
```suggestion
        if (vehicleStats) {
	  for (const vehicle of vehicleStats) {
          const existingTank = allPlayerStats.tanks.find((tank) => tank.wotId === vehicle.wotId);
      
          if (existingTank) {
            for (const statName of statsNames) {
              existingTank.regular[statName] += vehicle.regular[statName];
            }
            for (const achievementName of achievementsName) {
              existingTank.mastery[achievementName] += vehicle.mastery[achievementName] || 0;
            }
            existingTank.battleLifeTime += vehicle.battleLifeTime;
          } else {
            allPlayerStats.tanks.push(vehicle);
          }
        }
      }

```